### ISSUE TITLE
Sample on how to process each message (in JSON)?

### ISSUE THREAD

**Issue by olavt** [Azure/azure-event-hubs-python#66]
I'm following this example to process messages received from Event Hub:

https://github.com/Azure/azure-event-hubs-python/blob/master/examples/eph.py

But, the code does show any code on how to retrieve each message. How would one do this to get to the Json object?

I have tried the following:

async def process_events_async(self, context, messages):
        """
        Called by the processor host when a batch of events has arrived.
        This is where the real work of the event processor is done.
        :param context: Information about the partition
        :type context: ~azure.eventprocessorhost.PartitionContext
        :param messages: The events to be processed.
        :type messages: list[~azure.eventhub.common.EventData]
        """

        for message in messages:
            message_content = json.loads(message.body)
            print (message.body)

        logger.info("Events processed {}".format(context.sequence_number))
        await context.checkpoint_async()

But, I get this Python error on the json.loads statement:

"2018-09-20 23:13:12,484 azure        ERROR    Event Processor Error TypeError("the JSON object must be str, bytes or bytearray, not 'generator'",)"

Could we add to the example some code, which shows how to process the messages in Json format?

**Comment by annatisch** [comment]
Thanks @olavt!

The data in an event hub message can be in multiple sections, so we return it as a generator. Given that most people are using a sigle-section string in the body, I'm considering adding an easier way to access it....

In the meantime, you should be able to access the message content by iterating over the message.body.

For example:
```
for section in message.body:
    content = json.loads(section)
    print(content)
```

Or, if your data has multiple sections:
```
content = b"".join(message.body)
```

Or you could inspect any/all sections in the body as a list:
```
message_body = list(message.body)
print(message_body)
content = json.loads(message_body[0])
```

I hope this helps! :)

**Comment by annatisch** [comment]
Hi @olavt,

I have added two new convenience methods to EventData, `body_as_str()` and `body_as_json()` to help make this scenario simpler. I've also added demonstration of `body_as_str()` to the recv.py sample.

You can see me changes in the PR here: #65 

I'm hoping to release these changes (v1.1.0) this week.

**Comment by olavt** [comment]
@annatisch Yes, I now got it to work as I wanted. Thanks!

**Comment by annatisch** [comment]
I have now published v1.1.0 that includes the extra convenience functions for processing the event data.
I've also updated the recv.py sample to show using them.

Thanks!
